File: prql-dotnet/PrqlCompiler/Result.cs
using System.Collections.Generic;
using System.Linq;

namespace Prql.Compiler
{
    /// <summary>
    /// Result of compilation.
    /// </summary>
    public class Result
    {
        private readonly IReadOnlyCollection<Message> _messages;

        internal Result(string output, IEnumerable<Message> messages)
        {
            Output = output;
            _messages = messages.ToList().AsReadOnly();
        }

        /// <summary>
        /// @var string
        /// </summary>
        public string Output { get; }

        /// <summary>
        /// @var array<Message>
        /// </summary>
        public IReadOnlyCollection<Message> Messages => _messages;
    }
}

File: bindings/prql-php/src/Result.php
<?php

declare(strict_types=1);

namespace Prql\Compiler;

/**
 * Result of compilation.
 */
final class Result
{
    /**
     * @var string
     */
    public string $output;

    /**
     * @var array<Message>
     */
    public array $messages;
}
